src/mixer.c

```diff
@@ -0,0 +1,176 @@
+#include <stdint.h>
+
+#include <breezystm32.h>
+
+#include "mixer.h"
+
+int32_t _GPIO_outputs[8];
+int32_t _outputs[8];
+command_t _command;
+output_type_t _GPIO_output_type[8];
+
+static mixer_t quadcopter_plus_mixing = {
+  {M, M, M, M, 0, 0, 0, 0}, // output_type
+
+  { 1000, 1000,  1000, 1000, 0, 0, 0, 0}, // F Mix
+  { 0,   -1000,  1000, 0,    0, 0, 0, 0}, // X Mix
+  {-1000, 0,     0,    1000, 0, 0, 0, 0}, // Y Mix
+  {-1000, 1000,  1000,-1000, 0, 0, 0, 0}  // Z Mix
+};
+
+
+static mixer_t quadcopter_x_mixing = {
+  {M, M, M, M, 0, 0, 0, 0}, // output_type
+
+  { 1000, 1000, 1000, 1000,  0, 0, 0, 0}, // F Mix
+  {-1000,-1000, 1000, 1000,  0, 0, 0, 0}, // X Mix
+  {-1000, 1000,-1000, 1000,  0, 0, 0, 0}, // Y Mix
+  { 1000,-1000,-1000, 1000,  0, 0, 0, 0}  // Z Mix
+};
+
+static mixer_t fixedwing_mixing = {
+  {M, S, S, S, 0, 0, 0, 0},
+
+  {1000, 0,    0,    0,    0, 0, 0, 0}, // F Mix
+  {0,    1000, 0,    0,    0, 0, 0, 0}, // X Mix
+  {0,    0,    1000, 0,    0, 0, 0, 0}, // Y Mix
+  {0,    0,    0,    1000, 0, 0, 0, 0}  // Z Mix
+};
+
+static mixer_t tricopter_mixing = {
+  {M, M, M, S, 0, 0, 0, 0},
+
+  {1000,  1000, 1000, 0,    0, 0, 0, 0}, // F Mix
+  {-1000, 1000, 0,    0,    0, 0, 0, 0}, // X Mix
+  {-667,  -667, 1333, 0,    0, 0, 0, 0}, // Y Mix
+  {0,     0,    0,    1000, 0, 0, 0, 0}  // Z Mix
+};
+
+static mixer_t Y6_mixing = {
+  {M, M, M, M, M, M, 0, 0},
+  { 1000, 1000, 1000, 1000, 1000, 1000, 0, 0}, // F Mix
+  { 0,   -1000, 1000, 0,   -1000, 1000, 0, 0}, // X Mix
+  {-1300,  667,  667,-1300,  667,  667, 0, 0}, // Y Mix
+  {-1000, 1000, 1000, 1000,-1000,-1000, 0, 0}  // Z Mix
+};
+
+static mixer_t mixer_to_use;
+
+static mixer_t *array_of_mixers[5] = {
+  &quadcopter_plus_mixing,
+  &quadcopter_x_mixing,
+  &tricopter_mixing,
+  &Y6_mixing,
+  &fixedwing_mixing
+};
+
+
+
+void init_mixing()
+{
+  // We need a better way to choosing the mixer
+  mixer_to_use = *array_of_mixers[QUADCOPTER_X];
+
+  for(int8_t i=0; i<8; i++)
+  {
+    _outputs[i] = 0;
+    _GPIO_outputs[i] = 0;
+    _GPIO_output_type[i] = 0;
+  }
+  _command.F = 0;
+  _command.x = 0;
+  _command.y = 0;
+  _command.z = 0;
+}
+
+
+void write_motor(uint8_t index, int32_t value){
+  /** TODO:
+   * reverse the PWM -> mRad conversion in RC
+   */
+
+  if (value > 1000)
+  {
+    value = 1000;
+  }
+  else if (value < 0)
+  {
+    value = 0;
+  }
+  pwmWriteMotor(index, value+2000);
```

Comment:
Should this be value+1000 so that the resulting PWM is in the 1000us to 2000us range?